src/sdk/config_fs.js

```diff
@@ -265,104 +450,217 @@ class ConfigFS {
-        const account_config_path = this.get_account_path_by_name(account_name);
-        await native_fs_utils.update_config_file(this.fs_context, this.accounts_dir_path,
-            account_config_path, JSON.stringify(account_new_data));
-
-        if (new_access_keys_to_link.length > 0) {
-            for (const access_keys of new_access_keys_to_link) {
-                const new_access_key_path = this.get_account_path_by_access_key(access_keys.access_key);
-                const account_config_relative_path = this.get_account_relative_path_by_name(account_name);
-                await nb_native().fs.symlink(this.fs_context, account_config_relative_path, new_access_key_path);
-            }
-        }
-
-        if (old_access_keys_to_remove.length > 0) {
-            for (const access_key of old_access_keys_to_remove) {
-                const cur_access_key_path = this.get_account_path_by_access_key(access_key);
-                await nb_native().fs.unlink(this.fs_context, cur_access_key_path);
-            }
+    async update_account_config_file(account_new_data, options = {}) {
+        const { _id, name, owner = undefined } = account_new_data;
+        const data_string = JSON.stringify(account_new_data);
+        const account_path = this.get_identity_path_by_id(_id);
+        const account_dir_path = this.get_identity_dir_path_by_id(_id);
+        await native_fs_utils.update_config_file(this.fs_context, account_dir_path, account_path, data_string);
+
+        if (options.old_name) {
+            await this.link_account_name_index(_id, name, owner);
+            await this.unlink_account_name_index(options.old_name, account_path);
         }
+        await this.link_access_keys_index(_id, options.new_access_keys_to_link);
+        await this.unlink_access_keys_indexes(options.access_keys_to_delete, account_path);
     }
 
     /**
      * delete_account_config_file deletes account config file
-     * if access_keys_to_delete is an array that contains at least 1 item -
-     * unlink all items in access_keys_to_delete
+     * 1. unlink /access_keys/access_key if access_keys_to_delete is an array that contains at least 1 item -
+     *    unlink all item in access_keys_to_delete
+     * 2. unlink /root_accounts/account_name
+     * 3. delete /identities/account_id/identity.json
+     * 4. delete /identities/account_id/ folder
+     * @param {Object} data
+     * @returns {Promise<void>}
+     */
+    async delete_account_config_file(data) {
+        const { _id, name, access_keys = [] } = data;
+        const account_id_config_path = this.get_identity_path_by_id(_id);
+        const account_dir_path = this.get_identity_dir_path_by_id(_id);
+
+        await this.unlink_access_keys_indexes(access_keys, account_id_config_path);
+        await this.unlink_account_name_index(name, account_id_config_path);
+        await native_fs_utils.delete_config_file(this.fs_context, account_dir_path, account_id_config_path);
```

Comment:
If the file somehow doesn't exist will it throw an error? (because we don't want that and want to continue to the directory deletion).